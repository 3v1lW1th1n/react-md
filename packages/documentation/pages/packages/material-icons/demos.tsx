import MaterialIcons from "components/Demos/MaterialIcons";

export default MaterialIcons;
