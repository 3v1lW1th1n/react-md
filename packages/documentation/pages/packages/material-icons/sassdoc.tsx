import React from "react";
import { TextContainer } from "@react-md/typography";

import Heading from "components/Heading";

export default () => (
  <TextContainer>
    <Heading id="sassdoc" level={2}>
      SassDoc
    </Heading>
    <Heading id="coming-soon" level={4}>
      Coming Soon?
    </Heading>
  </TextContainer>
);
