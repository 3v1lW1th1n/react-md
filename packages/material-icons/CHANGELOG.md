# v2.0.0

This package is completely new for v2 and introduced the new 1864 icons (932
font icons and 932 svg icons).
