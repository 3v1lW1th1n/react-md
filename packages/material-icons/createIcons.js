#!/usr/bin/env node

const _ = require('lodash');
const fs = require('fs-extra');
const path = require('path');
const glob = require('glob');
const unzipper = require('unzipper');
const download = require('download');
const commander = require('commander');
const Bluebird = require('bluebird');
const execSync = require('child_process').execSync;

const readFile = Bluebird.promisify(fs.readFile);

commander
  .usage('<version> [options]')
  .option('-d, --download-only [downloadOnly]', 'This will only download and unzip the material-design-icons into the temp directory')
  .option('-i, --icons-only [iconsOnly]', 'This will only find and replace all icons within the temp directory for material-design-icons into the svg directory.')
  .option('-u, --update-only [updateOnly]', 'This will just update the generated component files only instead of downloading and finding icons from the source.')
  .option('-c, --clean [clean]', 'This will remove all the temp files created by this script')
  .option('--no-cleanup', 'This will prevent the script from automatically removing all the temp files if no other options are provided')
  .parse(process.argv);

let [version] = commander.args;
if (!version) {
  // the current latest version
  version = '3.0.1';
}

const fileName = `material-design-icons-${version}.zip`;
const downloadUrl = `https://github.com/google/material-design-icons/archive/${version}.zip`;
// const downloadUrl = `https://github.com/google/material-design-icons/archive/${version}/${fileName}`;
const src = path.join(process.cwd(), 'src');
const temp = path.join(process.cwd(), 'temp');
const svgs = path.join(process.cwd(), 'svgs');
const mainIndex = path.join(src, 'index.ts');


const EXLCUDE_REGEX = /(signal_wifi_[0-3])|(battery_(charging_)?\d)|((cellular|internet)_[^4]_bar)/;

async function downloadSource() {
  await fs.remove(temp);
  await fs.ensureDir(temp),
  console.log(`Downloading: '${downloadUrl}'`);
  await download(downloadUrl, temp)

  console.log(`Unzipping to '${path.join(temp, fileName.replace(/\.zip$/, ''))}'...`);
  fs.createReadStream(path.join(temp, fileName))
    .pipe(unzipper.Extract({ path: temp }))
  console.log('Done unzipping!');
}

function findAndCopyIcons() {
  return new Promise(async (resolve, reject) => {
    console.log('Removing old svgs folder...');
    await fs.remove(svgs);
    await fs.ensureDir(svgs);

    console.log(`Finding all icons and copying into '${svgs}'...`);
    glob('**/production/*_24px.svg', { root: temp }, async (err, matches) => {
      if (err) {
        reject(err);
        return;
      }

      const fileNames = [];
      await Promise.all(matches.reduce((promises, svgPath) => {
        const fileName = svgPath.substring(svgPath.lastIndexOf('/') + 1)
          .substring('ic_'.length)
          .replace(/_24px/, '');

        if (!EXLCUDE_REGEX.test(fileName) && fileNames.indexOf(fileName) === -1) {
          fileNames.push(fileName);
          promises.push(fs.copy(svgPath, path.join(svgs, fileName)));
        }

        return promises;
      }, []));

      console.log(`Copied ${fileNames.length} unique icons into '${svgs}'!`);
      resolve();
    });
  });
}

function toPascalCase(fileName) {
  if (fileName.match(/^[0-9]/)) {
    const [first, second, ...remaining] = fileName.split('_');
    fileName = `${second}_${first}${remaining.length ? `_${remaining.join('_')}` : ''}`;
  }

  return _.upperFirst(_.camelCase(fileName));
}

function createIconFile(componentName, children, iconType) {
  return `/* tslint:disable */
// This is a generated file from running the "createIcons" script. This file should not be updated manually.
import * as React from "react";

import { ${iconType}Icon, I${iconType}IconProps } from "@react-md/icon";

const ${componentName}${iconType}Icon: React.SFC<I${iconType}IconProps> = props => <${iconType}Icon {...props}>${children}</${iconType}Icon>;

export default ${componentName}${iconType}Icon;

`;
}

// kind of hacky, but each icon starts and ends the same way right now..
const SVG_ICON_PREFIX = '<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">';
const SVG_ICON_SUFFIX = '</svg>';

async function parseSVGFileAndCreateComponents(svgFilePath, componentName, iconName, svgIconFile, fontIconFile) {
  const svg = await readFile(path.join(process.cwd(), svgFilePath), 'utf8');
  const contents = svg.substring(SVG_ICON_PREFIX.length, svg.length - SVG_ICON_SUFFIX.length)
    .replace(/fill-opacity/g, 'fillOpacity');

  await Promise.all([
    fs.outputFile(svgIconFile, createIconFile(componentName, contents, 'SVG')),
    fs.outputFile(fontIconFile, createIconFile(componentName, iconName, 'Font')),
  ]);
}

async function createIndexFile(components) {
  const contents = `// This is a generated file from running the "createIcons" script. This file should not be updated manually.
${components.reduce((s, c) => `${s ? `${s}\n` : ''}export { default as ${c} } from "./${c}";`, '')}
`;

  return fs.outputFile(mainIndex, contents);
}

async function createComponentFiles() {
  return new Promise((resolve, reject) => {
    console.log(`Finding all svgs within the '${svgs}' folder...`);
    glob('svgs/*.svg', async (err, svgFiles) => {
      if (err) {
        reject(err);
        return;
      }

      await fs.remove(src);
      await fs.ensureDir(src);

      console.log('Creating all the SVGIcon and FontIcon components...');
      const components = [];
      await Promise.all(svgFiles.map((svgFilePath) => {
        const iconName = svgFilePath.replace(/.+\//, '').replace(/\.svg$/, '');
        const componentName = toPascalCase(iconName);
        const svgIconName = `${componentName}SVGIcon`;
        const fontIconName = `${componentName}FontIcon`;
        const svgIconFile = path.join(src, `${svgIconName}.tsx`);
        const fontIconFile = path.join(src, `${fontIconName}.tsx`);
        components.push(svgIconName, fontIconName);

        return parseSVGFileAndCreateComponents(svgFilePath, componentName, iconName, svgIconFile, fontIconFile);
      }));
      console.log('Updating the main index file to include all the components...');
      await createIndexFile(components);

      console.log('Running prettier on generated files...');
      execSync('npm run prettier');
      console.log('Done!');
    });
  });
}

async function cleanFiles() {
  console.log(`Removing the '${temp}' folder...`);
  await fs.remove(temp);
  console.log('Done!');
}

(async function run() {
  const { updateOnly, downloadOnly, iconsOnly, clean, cleanup } = commander;
  if (clean) {
    await cleanFiles();
  } else if (downloadOnly) {
    await downloadSource();
  } else if (iconsOnly) {
    await findAndCopyIcons();
  } else if (updateOnly) {
    await createComponentFiles();
  } else {
    await downloadSource();
    await findAndCopyIcons();
    await createComponentFiles();

    if (cleanup) {
      await cleanFiles();
    }
  }
})();
