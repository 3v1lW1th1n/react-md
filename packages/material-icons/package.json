{
  "name": "@react-md/material-icons",
  "version": "2.0.0-alpha.9",
  "description": "This package is just a simple wrapper for using material icons within react-md as either font icons or svg icons.",
  "scripts": {
    "clean": "dev-utils clean bin",
    "executable": "chmod +x bin/cli.js",
    "compile": "tsc -p tsconfig.cli.json",
    "build-cli": "npm-run-all compile executable",
    "watcher": "chokidar \"bin/cli.js\" -c \"npm run executable\"",
    "watch": "npm-run-all -p \"compile -- -w\" watcher",
    "build": "dev-utils build"
  },
  "main": "./lib/index.js",
  "module": "./es/index.js",
  "types": "./types/index.d.ts",
  "files": [
    "es/*",
    "lib/*",
    "types/*",
    "src/*",
    "svgs/*"
  ],
  "author": "Mikkel Laursen <mlaursen03@gmail.com>",
  "repository": {
    "type": "git",
    "url": "https://github.com/mlaursen/react-md",
    "directory": "packages/material-icons"
  },
  "bugs": {
    "url": "https://github.com/mlaursen/react-md/issues"
  },
  "keywords": [
    "react-md",
    "material design",
    "react",
    "component",
    "components",
    "icons",
    "svg icon",
    "font icon",
    "material design icons"
  ],
  "license": "MIT",
  "devDependencies": {
    "@react-md/icon": "^2.0.0-alpha.9",
    "download": "^7.1.0",
    "unzipper": "^0.10.4"
  },
  "peerDependencies": {
    "@react-md/icon": "^2.0.0-alpha.0",
    "react": ">= 16.8"
  },
  "publishConfig": {
    "access": "public"
  }
}
