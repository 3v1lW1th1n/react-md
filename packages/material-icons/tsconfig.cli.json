{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./bin",
    "rootDir": "scripts",
    "target": "es5",
    "module": "commonjs"
  },
  "include": [
    "scripts"
  ]
}
