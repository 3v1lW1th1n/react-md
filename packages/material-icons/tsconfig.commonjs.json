{"extends":"../../tsconfig.commonjs.json","compilerOptions":{"outDir":"./lib","rootDir":"src","declaration":false,"target":"es5"},"include":["src"],"exclude":["**/__tests__/*"]}
