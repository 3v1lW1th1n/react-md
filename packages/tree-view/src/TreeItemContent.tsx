import * as React from "react";
import cn from "classnames";
import { StatesConsumer } from "@react-md/states";
import { IListItemBaseProps, ListItemText, ListItemLeftIcon, ListItemRightIcon } from "@react-md/list";

import { ITreeItemDefaultProps } from "./TreeItem";

export interface ITreeItemContentBaseProps extends IListItemBaseProps {
  // TODO: Learn how to correctly type this
  linkComponent?: React.ReactType;
  medium?: boolean;
}

export type ITreeItemContentDivProps = ITreeItemContentBaseProps & React.HTMLAttributes<HTMLDivElement>;
export interface ITreeItemContentLinkProps extends ITreeItemContentBaseProps, React.HTMLAttributes<HTMLAnchorElement> {
  [key: string]: any;
  linkComponent?: React.ReactType;
}

export type ITreeItemContentProps = ITreeItemContentDivProps | ITreeItemContentLinkProps;
export interface ITreeItemContentDefaultProps {
  medium: boolean;
  linkComponent: React.ReactType;
}
export type TreeItemContentWithDefaultProps = ITreeItemContentProps &
  ITreeItemContentDefaultProps &
  ITreeItemDefaultProps;

const TreeItemContent: React.SFC<ITreeItemContentProps> = providedProps => {
  const {
    className: propClassName,
    onMouseDown,
    onMouseUp,
    onTouchStart,
    onTouchEnd,
    onKeyDown,
    onKeyUp,
    linkComponent,
    leftIcon,
    rightIcon,
    forceIconWrap,
    children,
    medium,
    ...props
  } = providedProps as TreeItemContentWithDefaultProps;

  const className = cn(
    "rmd-tree-item__content",
    {
      "rmd-tree-item__content--link": !!linkComponent,
    },
    "rmd-list-item rmd-list-item--stateful",
    {
      "rmd-list-item--medium": medium && !!linkComponent,
    },
    propClassName
  );

  const content = (
    <ListItemLeftIcon icon={leftIcon} forceIconWrap={forceIconWrap}>
      <ListItemRightIcon icon={rightIcon} forceIconWrap={forceIconWrap}>
        <ListItemText>{children}</ListItemText>
      </ListItemRightIcon>
    </ListItemLeftIcon>
  );

  if (!linkComponent) {
    const divProps = props as ITreeItemContentDivProps;
    return (
      <div {...divProps} className={className}>
        {content}
      </div>
    );
  }

  return React.createElement(
    linkComponent,
    {
      ...props,
      className,
    },
    content
  );
};

TreeItemContent.defaultProps = {
  medium: false,
} as ITreeItemContentDefaultProps;

export default TreeItemContent;
