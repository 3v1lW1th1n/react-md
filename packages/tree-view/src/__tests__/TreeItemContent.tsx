import * as React from "react";
import renderer from "react-test-renderer";
import { HomeSVGIcon } from "@react-md/material-icons";

import TreeItemContent from "../TreeItemContent";

// TODO: write more tests when enzyme supports new React context api
describe("TreeItemContent", () => {
  it("should render correctly with the provided props", () => {
    expect(renderer.create(<TreeItemContent>Content!</TreeItemContent>).toJSON()).toMatchSnapshot();
    expect(
      renderer.create(<TreeItemContent leftIcon={<HomeSVGIcon />}>Content with icon</TreeItemContent>).toJSON()
    ).toMatchSnapshot();
    expect(
      renderer.create(<TreeItemContent rightIcon={<HomeSVGIcon />}>Content with icon</TreeItemContent>).toJSON()
    ).toMatchSnapshot();
  });
});
