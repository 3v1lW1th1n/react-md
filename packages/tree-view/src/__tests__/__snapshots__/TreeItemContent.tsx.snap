// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeItemContent should render correctly with the provided props 1`] = `
<div
  className="rmd-tree-item__content rmd-list-item rmd-list-item--stateful"
>
  <span
    className="rmd-list-item__text"
  >
    Content!
  </span>
</div>
`;

exports[`TreeItemContent should render correctly with the provided props 2`] = `
<div
  className="rmd-tree-item__content rmd-list-item rmd-list-item--stateful"
>
  <svg
    className="rmd-icon rmd-icon--svg rmd-list-item__icon rmd-list-item__icon--before"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
    />
  </svg>
  <span
    className="rmd-list-item__text"
  >
    Content with icon
  </span>
</div>
`;

exports[`TreeItemContent should render correctly with the provided props 3`] = `
<div
  className="rmd-tree-item__content rmd-list-item rmd-list-item--stateful"
>
  <span
    className="rmd-list-item__text"
  >
    Content with icon
  </span>
  <svg
    className="rmd-icon rmd-icon--svg rmd-list-item__icon rmd-list-item__icon--after"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
    />
  </svg>
</div>
`;
